Review of DM-8156: add support for Slurm to allocateNodes.py

hsinfang opened the pull request.

hsinfang commented on python/lsst/ctrl/execute/allocator.py:
```diff
@@ -86,11 +86,11 @@ def __init__(self, platform, opts, configFileName):
                 user_home = os.getenv('HOME')
 
         if user_name is None:
-            raise RuntimeError("error: %s does not specify user name for platform %s" %
```
About line 82: would you consider changing or adding `self.platform == "verification":` and similar placeholders for the verification cluster platform? Seems convenient for the users.

hsinfang replied:
Why deleting the first `%s`?

srp3rd replied:
The first %s was missing.  Added "==" per request.

hsinfang commented:
Before merging, may you please rewrite the commit message in the imperative tense and squash as you see fit? (e.g. "more refactoring" is not very clear what the commit is about) 

https://developer.lsst.io/processes/workflow.html#git-commit-message-best-practices

hsinfang commented on python/lsst/ctrl/execute/namedClassFactory.py:
```diff
@@ -0,0 +1,54 @@
+#
+# LSST Data Management System
+# Copyright 2008, 2009, 2010 LSST Corporation.
```
year needs update
(and in a few other files as well)

srp3rd replied:
Updated in all files.

hsinfang commented on python/lsst/ctrl/execute/namedClassFactory.py:
```diff
@@ -0,0 +1,54 @@
+    Parameters
+    ----------
+    name : `str`
+        the fully qualified nname of an object
```
nname -> name

hsinfang commented on python/lsst/ctrl/execute/allocator.py:
```diff
@@ -139,37 +141,34 @@ def createUniqueIdentifier(self):
-
-    def loadPbs(self, name):
-        """Loads all values from configuration and command line overrides into
+        # 11/17/16 - srp XXX
```
Did you forget to delete this?

srp3rd replied:
yes, I did forget.  Thanks

srp3rd committed:
add additional configuration elements

Add "scheduler" and "setup_using" elements to configs
to specify which scheduler to use and how to run the eups setups.

srp3rd committed:
Create plugins and refactor

Create plugins for pbs and slurm (and a class loader for
the named plugins) which now use Allocator as a base class,  and
refactor how allocateNodes calls
the plugins.

srp3rd committed:
Change how eups setup environment is specified

Setups used to be done on a per-platform basis, and have been changed
to be specified within the configuration so that no platforms will be
hard-coded.  Also fixed name substituion for LOCAL_SCRATCH

srp3rd committed:
Add NODE_SET for workflow configuration

Adds specification of NODE_SET for keywords to be used for value
subtitution.

srp3rd committed:
Update unit tests

Update exisiting unit tests to test addition features.
Added unit tests that exercise the addition of the
slurm plugin.

srp3rd committed:
Updated copyright year to 2016

srp3rd commented:
"Before merging, may you please rewrite the commit message in the imperative tense and squash as you see fit? (e.g. "more refactoring" is not very clear what the commit is about)"

I updated these, and fixed the copyright year.
